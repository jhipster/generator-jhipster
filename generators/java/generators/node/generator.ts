/**
 * Copyright 2013-2025 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project, see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import chalk from 'chalk';
import type { ExecaError } from 'execa';
import BaseApplicationGenerator from '../../../base-application/index.js';

export default class NodeGenerator extends BaseApplicationGenerator {
  async beforeQueue() {
    if (!this.fromBlueprint) {
      await this.composeWithBlueprints();
    }

    if (!this.delegateToBlueprint) {
      await this.dependsOnBootstrapApplication();
      await this.dependsOnJHipster('jhipster:java:build-tool');
    }
  }

  get composing() {
    return this.asComposingTaskGroup({
      async compose() {
        const { buildTool } = this.jhipsterConfigWithDefaults;
        if (buildTool === 'maven') {
          await this.composeWithJHipster('jhipster:maven:frontend-plugin');
        } else if (buildTool === 'gradle') {
          await this.composeWithJHipster('jhipster:gradle:node-gradle');
        }
      },
    });
  }

  get [BaseApplicationGenerator.COMPOSING]() {
    return this.delegateTasksToBlueprint(() => this.composing);
  }

  get preparing() {
    return this.asPreparingTaskGroup({
      async javaNodeBuildPaths({ application }) {
        const { buildToolMaven, srcMainWebapp, javaNodeBuildPaths, clientDistDir } = application;

        javaNodeBuildPaths.push(srcMainWebapp, 'package-lock.json', 'package.json');
        if (buildToolMaven) {
          // Gradle throws an error if the directory does not exist
          javaNodeBuildPaths.push(clientDistDir!);
        }
      },
    });
  }

  get [BaseApplicationGenerator.PREPARING]() {
    return this.delegateTasksToBlueprint(() => this.preparing);
  }

  get postPreparing() {
    return this.asPostPreparingTaskGroup({
      sortBuildFiles({ application }) {
        const { javaNodeBuildPaths } = application;
        if (javaNodeBuildPaths) {
          const files = [...new Set(javaNodeBuildPaths)];
          javaNodeBuildPaths.splice(0, javaNodeBuildPaths.length, ...files.sort());
        }
      },
      useNpmWrapper({ application }) {
        if (application.useNpmWrapper) {
          this.useNpmWrapperInstallTask();
        }
      },
    });
  }

  get [BaseApplicationGenerator.POST_PREPARING]() {
    return this.delegateTasksToBlueprint(() => this.postPreparing);
  }

  get writing() {
    return this.asWritingTaskGroup({
      async writing({ application }) {
        await this.writeFiles({
          blocks: [
            {
              condition: (ctx: any) => ctx.useNpmWrapper,
              templates: ['npmw', 'npmw.cmd'],
            },
          ],
          context: application,
        });
      },
    });
  }

  get [BaseApplicationGenerator.WRITING]() {
    return this.delegateTasksToBlueprint(() => this.writing);
  }

  useNpmWrapperInstallTask() {
    this.setFeatures({
      customInstallTask: async (preferredPm, defaultInstallTask) => {
        const buildTool = this.jhipsterConfigWithDefaults.buildTool;
        if ((preferredPm && preferredPm !== 'npm') || this.jhipsterConfig.skipClient || (buildTool !== 'gradle' && buildTool !== 'maven')) {
          await defaultInstallTask();
          return;
        }

        const npmCommand = process.platform === 'win32' ? 'npmw' : './npmw';
        try {
          await this.spawn(npmCommand, ['install'], { preferLocal: true });
        } catch (error: unknown) {
          this.log.error(
            chalk.red(`Error executing '${npmCommand} install', please execute it yourself. (${(error as ExecaError).shortMessage})`),
          );
        }
      },
    });
  }
}
