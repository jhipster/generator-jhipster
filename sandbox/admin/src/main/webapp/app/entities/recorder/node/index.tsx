import React from 'react';
import { Switch } from 'react-router-dom';

import ErrorBoundaryRoute from 'app/shared/error/error-boundary-route';

import Node from './node';
import NodeDetail from './node-detail';
import NodeUpdate from './node-update';
import NodeDeleteDialog from './node-delete-dialog';

const Routes = ({ match }) => (
  <>
    <Switch>
      <ErrorBoundaryRoute exact path={`${match.url}/new`} component={NodeUpdate} />
      <ErrorBoundaryRoute exact path={`${match.url}/:id/edit`} component={NodeUpdate} />
      <ErrorBoundaryRoute exact path={`${match.url}/:id`} component={NodeDetail} />
      <ErrorBoundaryRoute path={match.url} component={Node} />
    </Switch>
    <ErrorBoundaryRoute exact path={`${match.url}/:id/delete`} component={NodeDeleteDialog} />
  </>
);

export default Routes;
