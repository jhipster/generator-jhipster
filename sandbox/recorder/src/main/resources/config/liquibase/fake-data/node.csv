id;name;description;time_to_live
1;1080p Taiwan;content;19346
2;strategic;synthesizing system transition;76082
3;object-oriented Qatari Soft;application success;62180
4;Tools;haptic;95444
5;bifurcated Sharable Canyon;web-readiness;90738
6;bandwidth;Loan Tasty synergies;85820
7;Palestinian Rubber;Intelligent Enhanced Chief;89121
8;Managed AGP Organic;Functionality;43141
9;blue Baby Home;Plastic Fish lavender;17232
10;Berkshire Shirt supply-chains;SCSI motivating didactic;33060
