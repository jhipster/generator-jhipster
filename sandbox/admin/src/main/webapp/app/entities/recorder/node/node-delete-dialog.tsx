import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router-dom';
import { Modal, ModalHeader, ModalBody, ModalFooter, Button } from 'reactstrap';
import { Translate } from 'react-jhipster';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { IRootState } from 'app/shared/reducers';
import { getEntity, deleteEntity } from './node.reducer';

export interface INodeDeleteDialogProps extends StateProps, DispatchProps, RouteComponentProps<{ id: string }> {}

export const NodeDeleteDialog = (props: INodeDeleteDialogProps) => {
  useEffect(() => {
    props.getEntity(props.match.params.id);
  }, []);

  const handleClose = () => {
    props.history.push('/node');
  };

  useEffect(() => {
    if (props.updateSuccess) {
      handleClose();
    }
  }, [props.updateSuccess]);

  const confirmDelete = () => {
    props.deleteEntity(props.nodeEntity.id);
  };

  const { nodeEntity } = props;
  return (
    <Modal isOpen toggle={handleClose}>
      <ModalHeader toggle={handleClose} data-cy="nodeDeleteDialogHeading">
        <Translate contentKey="entity.delete.title">Confirm delete operation</Translate>
      </ModalHeader>
      <ModalBody id="adminApp.recorderNode.delete.question">
        <Translate contentKey="adminApp.recorderNode.delete.question" interpolate={{ id: nodeEntity.id }}>
          Are you sure you want to delete this Node?
        </Translate>
      </ModalBody>
      <ModalFooter>
        <Button color="secondary" onClick={handleClose}>
          <FontAwesomeIcon icon="ban" />
          &nbsp;
          <Translate contentKey="entity.action.cancel">Cancel</Translate>
        </Button>
        <Button id="jhi-confirm-delete-node" data-cy="entityConfirmDeleteButton" color="danger" onClick={confirmDelete}>
          <FontAwesomeIcon icon="trash" />
          &nbsp;
          <Translate contentKey="entity.action.delete">Delete</Translate>
        </Button>
      </ModalFooter>
    </Modal>
  );
};

const mapStateToProps = ({ node }: IRootState) => ({
  nodeEntity: node.entity,
  updateSuccess: node.updateSuccess,
});

const mapDispatchToProps = { getEntity, deleteEntity };

type StateProps = ReturnType<typeof mapStateToProps>;
type DispatchProps = typeof mapDispatchToProps;

export default connect(mapStateToProps, mapDispatchToProps)(NodeDeleteDialog);
