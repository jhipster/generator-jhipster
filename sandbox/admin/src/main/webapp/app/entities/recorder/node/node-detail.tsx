import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { Link, RouteComponentProps } from 'react-router-dom';
import { Button, Row, Col } from 'reactstrap';
import { Translate } from 'react-jhipster';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { IRootState } from 'app/shared/reducers';
import { getEntity } from './node.reducer';
import { APP_DATE_FORMAT, APP_LOCAL_DATE_FORMAT } from 'app/config/constants';

export interface INodeDetailProps extends StateProps, DispatchProps, RouteComponentProps<{ id: string }> {}

export const NodeDetail = (props: INodeDetailProps) => {
  useEffect(() => {
    props.getEntity(props.match.params.id);
  }, []);

  const { nodeEntity } = props;
  return (
    <Row>
      <Col md="8">
        <h2 data-cy="nodeDetailsHeading">
          <Translate contentKey="adminApp.recorderNode.detail.title">Node</Translate>
        </h2>
        <dl className="jh-entity-details">
          <dt>
            <span id="id">
              <Translate contentKey="global.field.id">ID</Translate>
            </span>
          </dt>
          <dd>{nodeEntity.id}</dd>
          <dt>
            <span id="name">
              <Translate contentKey="adminApp.recorderNode.name">Name</Translate>
            </span>
          </dt>
          <dd>{nodeEntity.name}</dd>
          <dt>
            <span id="description">
              <Translate contentKey="adminApp.recorderNode.description">Description</Translate>
            </span>
          </dt>
          <dd>{nodeEntity.description}</dd>
          <dt>
            <span id="timeToLive">
              <Translate contentKey="adminApp.recorderNode.timeToLive">Time To Live</Translate>
            </span>
          </dt>
          <dd>{nodeEntity.timeToLive}</dd>
          <dt>
            <Translate contentKey="adminApp.recorderNode.parent">Parent</Translate>
          </dt>
          <dd>{nodeEntity.parent ? nodeEntity.parent.name : ''}</dd>
        </dl>
        <Button tag={Link} to="/node" replace color="info" data-cy="entityDetailsBackButton">
          <FontAwesomeIcon icon="arrow-left" />{' '}
          <span className="d-none d-md-inline">
            <Translate contentKey="entity.action.back">Back</Translate>
          </span>
        </Button>
        &nbsp;
        <Button tag={Link} to={`/node/${nodeEntity.id}/edit`} replace color="primary">
          <FontAwesomeIcon icon="pencil-alt" />{' '}
          <span className="d-none d-md-inline">
            <Translate contentKey="entity.action.edit">Edit</Translate>
          </span>
        </Button>
      </Col>
    </Row>
  );
};

const mapStateToProps = ({ node }: IRootState) => ({
  nodeEntity: node.entity,
});

const mapDispatchToProps = { getEntity };

type StateProps = ReturnType<typeof mapStateToProps>;
type DispatchProps = typeof mapDispatchToProps;

export default connect(mapStateToProps, mapDispatchToProps)(NodeDetail);
