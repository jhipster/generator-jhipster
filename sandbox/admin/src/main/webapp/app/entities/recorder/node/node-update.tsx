import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { Link, RouteComponentProps } from 'react-router-dom';
import { Button, Row, Col, Label } from 'reactstrap';
import { AvFeedback, AvForm, AvGroup, AvInput, AvField } from 'availity-reactstrap-validation';
import { Translate, translate } from 'react-jhipster';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { IRootState } from 'app/shared/reducers';

import { getEntities as getNodes } from 'app/entities/recorder/node/node.reducer';
import { IChannel } from 'app/shared/model/recorder/channel.model';
import { getEntities as getChannels } from 'app/entities/recorder/channel/channel.reducer';
import { IUserProfile } from 'app/shared/model/recorder/user-profile.model';
import { getEntities as getUserProfiles } from 'app/entities/recorder/user-profile/user-profile.reducer';
import { getEntity, updateEntity, createEntity, reset } from './node.reducer';
import { INode } from 'app/shared/model/recorder/node.model';
import { convertDateTimeFromServer, convertDateTimeToServer, displayDefaultDateTime } from 'app/shared/util/date-utils';
import { mapIdList } from 'app/shared/util/entity-utils';

export interface INodeUpdateProps extends StateProps, DispatchProps, RouteComponentProps<{ id: string }> {}

export const NodeUpdate = (props: INodeUpdateProps) => {
  const [isNew] = useState(!props.match.params || !props.match.params.id);

  const { nodeEntity, nodes, channels, userProfiles, loading, updating } = props;

  const handleClose = () => {
    props.history.push('/node');
  };

  useEffect(() => {
    if (isNew) {
      props.reset();
    } else {
      props.getEntity(props.match.params.id);
    }

    props.getNodes();
    props.getChannels();
    props.getUserProfiles();
  }, []);

  useEffect(() => {
    if (props.updateSuccess) {
      handleClose();
    }
  }, [props.updateSuccess]);

  const saveEntity = (event, errors, values) => {
    if (errors.length === 0) {
      const entity = {
        ...nodeEntity,
        ...values,
        parent: nodes.find(it => it.id.toString() === values.parentId.toString()),
      };

      if (isNew) {
        props.createEntity(entity);
      } else {
        props.updateEntity(entity);
      }
    }
  };

  return (
    <div>
      <Row className="justify-content-center">
        <Col md="8">
          <h2 id="adminApp.recorderNode.home.createOrEditLabel" data-cy="NodeCreateUpdateHeading">
            <Translate contentKey="adminApp.recorderNode.home.createOrEditLabel">Create or edit a Node</Translate>
          </h2>
        </Col>
      </Row>
      <Row className="justify-content-center">
        <Col md="8">
          {loading ? (
            <p>Loading...</p>
          ) : (
            <AvForm model={isNew ? {} : nodeEntity} onSubmit={saveEntity}>
              {!isNew ? (
                <AvGroup>
                  <Label for="node-id">
                    <Translate contentKey="global.field.id">ID</Translate>
                  </Label>
                  <AvInput id="node-id" type="text" className="form-control" name="id" required readOnly />
                </AvGroup>
              ) : null}
              <AvGroup>
                <Label id="nameLabel" for="node-name">
                  <Translate contentKey="adminApp.recorderNode.name">Name</Translate>
                </Label>
                <AvField
                  id="node-name"
                  data-cy="name"
                  type="text"
                  name="name"
                  validate={{
                    required: { value: true, errorMessage: translate('entity.validation.required') },
                  }}
                />
              </AvGroup>
              <AvGroup>
                <Label id="descriptionLabel" for="node-description">
                  <Translate contentKey="adminApp.recorderNode.description">Description</Translate>
                </Label>
                <AvField id="node-description" data-cy="description" type="text" name="description" />
              </AvGroup>
              <AvGroup>
                <Label id="timeToLiveLabel" for="node-timeToLive">
                  <Translate contentKey="adminApp.recorderNode.timeToLive">Time To Live</Translate>
                </Label>
                <AvField id="node-timeToLive" data-cy="timeToLive" type="string" className="form-control" name="timeToLive" />
              </AvGroup>
              <AvGroup>
                <Label for="node-parent">
                  <Translate contentKey="adminApp.recorderNode.parent">Parent</Translate>
                </Label>
                <AvInput id="node-parent" data-cy="parent" type="select" className="form-control" name="parentId">
                  <option value="" key="0" />
                  {nodes
                    ? nodes.map(otherEntity => (
                        <option value={otherEntity.id} key={otherEntity.id}>
                          {otherEntity.name}
                        </option>
                      ))
                    : null}
                </AvInput>
              </AvGroup>
              <Button tag={Link} id="cancel-save" data-cy="entityCreateCancelButton" to="/node" replace color="info">
                <FontAwesomeIcon icon="arrow-left" />
                &nbsp;
                <span className="d-none d-md-inline">
                  <Translate contentKey="entity.action.back">Back</Translate>
                </span>
              </Button>
              &nbsp;
              <Button color="primary" id="save-entity" data-cy="entityCreateSaveButton" type="submit" disabled={updating}>
                <FontAwesomeIcon icon="save" />
                &nbsp;
                <Translate contentKey="entity.action.save">Save</Translate>
              </Button>
            </AvForm>
          )}
        </Col>
      </Row>
    </div>
  );
};

const mapStateToProps = (storeState: IRootState) => ({
  nodes: storeState.node.entities,
  channels: storeState.channel.entities,
  userProfiles: storeState.userProfile.entities,
  nodeEntity: storeState.node.entity,
  loading: storeState.node.loading,
  updating: storeState.node.updating,
  updateSuccess: storeState.node.updateSuccess,
});

const mapDispatchToProps = {
  getNodes,
  getChannels,
  getUserProfiles,
  getEntity,
  updateEntity,
  createEntity,
  reset,
};

type StateProps = ReturnType<typeof mapStateToProps>;
type DispatchProps = typeof mapDispatchToProps;

export default connect(mapStateToProps, mapDispatchToProps)(NodeUpdate);
