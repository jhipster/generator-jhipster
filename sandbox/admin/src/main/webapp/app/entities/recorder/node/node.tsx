import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { Link, RouteComponentProps } from 'react-router-dom';
import { Button, Col, Row, Table } from 'reactstrap';
import { Translate } from 'react-jhipster';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { IRootState } from 'app/shared/reducers';
import { getEntities } from './node.reducer';
import { INode } from 'app/shared/model/recorder/node.model';
import { APP_DATE_FORMAT, APP_LOCAL_DATE_FORMAT } from 'app/config/constants';

export interface INodeProps extends StateProps, DispatchProps, RouteComponentProps<{ url: string }> {}

export const Node = (props: INodeProps) => {
  useEffect(() => {
    props.getEntities();
  }, []);

  const handleSyncList = () => {
    props.getEntities();
  };

  const { nodeList, match, loading } = props;
  return (
    <div>
      <h2 id="node-heading" data-cy="NodeHeading">
        <Translate contentKey="adminApp.recorderNode.home.title">Nodes</Translate>
        <div className="d-flex justify-content-end">
          <Button className="mr-2" color="info" onClick={handleSyncList} disabled={loading}>
            <FontAwesomeIcon icon="sync" spin={loading} />{' '}
            <Translate contentKey="adminApp.recorderNode.home.refreshListLabel">Refresh List</Translate>
          </Button>
          <Link to={`${match.url}/new`} className="btn btn-primary jh-create-entity" id="jh-create-entity" data-cy="entityCreateButton">
            <FontAwesomeIcon icon="plus" />
            &nbsp;
            <Translate contentKey="adminApp.recorderNode.home.createLabel">Create new Node</Translate>
          </Link>
        </div>
      </h2>
      <div className="table-responsive">
        {nodeList && nodeList.length > 0 ? (
          <Table responsive>
            <thead>
              <tr>
                <th>
                  <Translate contentKey="adminApp.recorderNode.id">ID</Translate>
                </th>
                <th>
                  <Translate contentKey="adminApp.recorderNode.name">Name</Translate>
                </th>
                <th>
                  <Translate contentKey="adminApp.recorderNode.description">Description</Translate>
                </th>
                <th>
                  <Translate contentKey="adminApp.recorderNode.timeToLive">Time To Live</Translate>
                </th>
                <th>
                  <Translate contentKey="adminApp.recorderNode.parent">Parent</Translate>
                </th>
                <th />
              </tr>
            </thead>
            <tbody>
              {nodeList.map((node, i) => (
                <tr key={`entity-${i}`} data-cy="entityTable">
                  <td>
                    <Button tag={Link} to={`${match.url}/${node.id}`} color="link" size="sm">
                      {node.id}
                    </Button>
                  </td>
                  <td>{node.name}</td>
                  <td>{node.description}</td>
                  <td>{node.timeToLive}</td>
                  <td>{node.parent ? <Link to={`node/${node.parent.id}`}>{node.parent.name}</Link> : ''}</td>
                  <td className="text-right">
                    <div className="btn-group flex-btn-group-container">
                      <Button tag={Link} to={`${match.url}/${node.id}`} color="info" size="sm" data-cy="entityDetailsButton">
                        <FontAwesomeIcon icon="eye" />{' '}
                        <span className="d-none d-md-inline">
                          <Translate contentKey="entity.action.view">View</Translate>
                        </span>
                      </Button>
                      <Button tag={Link} to={`${match.url}/${node.id}/edit`} color="primary" size="sm" data-cy="entityEditButton">
                        <FontAwesomeIcon icon="pencil-alt" />{' '}
                        <span className="d-none d-md-inline">
                          <Translate contentKey="entity.action.edit">Edit</Translate>
                        </span>
                      </Button>
                      <Button tag={Link} to={`${match.url}/${node.id}/delete`} color="danger" size="sm" data-cy="entityDeleteButton">
                        <FontAwesomeIcon icon="trash" />{' '}
                        <span className="d-none d-md-inline">
                          <Translate contentKey="entity.action.delete">Delete</Translate>
                        </span>
                      </Button>
                    </div>
                  </td>
                </tr>
              ))}
            </tbody>
          </Table>
        ) : (
          !loading && (
            <div className="alert alert-warning">
              <Translate contentKey="adminApp.recorderNode.home.notFound">No Nodes found</Translate>
            </div>
          )
        )}
      </div>
    </div>
  );
};

const mapStateToProps = ({ node }: IRootState) => ({
  nodeList: node.entities,
  loading: node.loading,
});

const mapDispatchToProps = {
  getEntities,
};

type StateProps = ReturnType<typeof mapStateToProps>;
type DispatchProps = typeof mapDispatchToProps;

export default connect(mapStateToProps, mapDispatchToProps)(Node);
