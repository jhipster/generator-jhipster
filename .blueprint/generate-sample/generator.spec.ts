import { before, describe, expect, it } from 'esmocha';
import { basename, dirname, join } from 'node:path';
import { fileURLToPath } from 'node:url';

import { defaultHelpers as helpers, runResult } from '../../lib/testing/index.ts';
import { shouldSupportFeatures } from '../../test/support/index.ts';

import Generator from './generator.ts';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const generator = basename(__dirname);

process.env.CI = 'true';

describe(`generator - ${generator}`, () => {
  shouldSupportFeatures(Generator);

  describe(`with ng-default`, () => {
    before(async () => {
      await helpers.runJHipster(join(__dirname, 'index.ts'), { useEnvironmentBuilder: true }).withArguments('ng-default').withOptions({
        sampleOnly: true,
      });
    });

    it('should match matrix value', () => {
      expect(runResult.getStateSnapshot()).toMatchSnapshot();
    });
  });

  describe(`with ng-default-additional (star jdl-entity)`, () => {
    before(async () => {
      await helpers
        .runJHipster(join(__dirname, 'index.ts'), { useEnvironmentBuilder: true })
        .withArguments('ng-default-additional')
        .withOptions({
          sampleOnly: true,
        });
    });

    it('should match matrix value', () => {
      expect(runResult.getStateSnapshot()).toMatchSnapshot();
    });
  });

  describe(`with vue-default-additional (specific jdl-entity)`, () => {
    before(async () => {
      await helpers
        .runJHipster(join(__dirname, 'index.ts'), { useEnvironmentBuilder: true })
        .withArguments('vue-default-additional')
        .withOptions({
          sampleOnly: true,
        });
    });

    it('should match matrix value', () => {
      expect(runResult.getStateSnapshot()).toMatchSnapshot();
    });
  });

  describe(`with ng-webflux-psql-additional (specific jdl-samples)`, () => {
    before(async () => {
      await helpers
        .runJHipster(join(__dirname, 'index.ts'), { useEnvironmentBuilder: true })
        .withArguments('ng-webflux-psql-additional')
        .withOptions({
          sampleOnly: true,
        });
    });

    it('should match matrix value', () => {
      expect(runResult.getStateSnapshot()).toMatchSnapshot();
    });
  });

  describe(`with daily-builds/ngx-oauth2 (daily-builds sample)`, () => {
    before(async () => {
      await helpers.runJHipster(join(__dirname, 'index.ts'), { useEnvironmentBuilder: true }).withArguments('ng-default').withOptions({
        sampleOnly: true,
      });
    });

    it('should match matrix value', () => {
      expect(runResult.getStateSnapshot()).toMatchSnapshot();
    });
  });
});
