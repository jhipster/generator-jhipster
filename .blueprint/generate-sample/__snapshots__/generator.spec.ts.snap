// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`generator - generate-sample with daily-builds/ngx-oauth2 (daily-builds sample) should match matrix value 1`] = `
{
  ".jhipster/BankAccount.json": {
    "stateCleared": "modified",
  },
  ".jhipster/Division.json": {
    "stateCleared": "modified",
  },
  ".jhipster/EntityWithDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/EntityWithPaginationAndDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/EntityWithServiceClassAndPagination.json": {
    "stateCleared": "modified",
  },
  ".jhipster/EntityWithServiceClassPaginationAndDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/EntityWithServiceImplAndDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/EntityWithServiceImplAndPagination.json": {
    "stateCleared": "modified",
  },
  ".jhipster/EntityWithServiceImplPaginationAndDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/FieldTestEntity.json": {
    "stateCleared": "modified",
  },
  ".jhipster/FieldTestEnumWithValue.json": {
    "stateCleared": "modified",
  },
  ".jhipster/FieldTestInfiniteScrollEntity.json": {
    "stateCleared": "modified",
  },
  ".jhipster/FieldTestMapstructAndServiceClassEntity.json": {
    "stateCleared": "modified",
  },
  ".jhipster/FieldTestPaginationEntity.json": {
    "stateCleared": "modified",
  },
  ".jhipster/FieldTestServiceClassAndJpaFilteringEntity.json": {
    "stateCleared": "modified",
  },
  ".jhipster/FieldTestServiceImplEntity.json": {
    "stateCleared": "modified",
  },
  ".jhipster/JpaFilteringOtherSide.json": {
    "stateCleared": "modified",
  },
  ".jhipster/JpaFilteringRelationship.json": {
    "stateCleared": "modified",
  },
  ".jhipster/Label.json": {
    "stateCleared": "modified",
  },
  ".jhipster/MapsIdChildEntityWithDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/MapsIdChildEntityWithoutDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/MapsIdGrandchildEntityWithDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/MapsIdGrandchildEntityWithoutDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/MapsIdParentEntityWithDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/MapsIdParentEntityWithoutDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/MapsIdUserProfileWithDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/Operation.json": {
    "stateCleared": "modified",
  },
  ".jhipster/Place.json": {
    "stateCleared": "modified",
  },
  ".jhipster/SuperMegaLargeTestEntity.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestCustomTableName.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestEntity.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestInfiniteScroll.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestManyRelPaginDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestManyToMany.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestManyToOne.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestMapstruct.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestOneToOne.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestPagination.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestServiceClass.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestServiceImpl.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestTwoRelationshipsSameEntity.json": {
    "stateCleared": "modified",
  },
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
}
`;

exports[`generator - generate-sample with ng-default should match matrix value 1`] = `
{
  ".jhipster/BankAccount.json": {
    "stateCleared": "modified",
  },
  ".jhipster/Division.json": {
    "stateCleared": "modified",
  },
  ".jhipster/EntityWithDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/EntityWithPaginationAndDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/EntityWithServiceClassAndPagination.json": {
    "stateCleared": "modified",
  },
  ".jhipster/EntityWithServiceClassPaginationAndDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/EntityWithServiceImplAndDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/EntityWithServiceImplAndPagination.json": {
    "stateCleared": "modified",
  },
  ".jhipster/EntityWithServiceImplPaginationAndDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/FieldTestEntity.json": {
    "stateCleared": "modified",
  },
  ".jhipster/FieldTestEnumWithValue.json": {
    "stateCleared": "modified",
  },
  ".jhipster/FieldTestInfiniteScrollEntity.json": {
    "stateCleared": "modified",
  },
  ".jhipster/FieldTestMapstructAndServiceClassEntity.json": {
    "stateCleared": "modified",
  },
  ".jhipster/FieldTestPaginationEntity.json": {
    "stateCleared": "modified",
  },
  ".jhipster/FieldTestServiceClassAndJpaFilteringEntity.json": {
    "stateCleared": "modified",
  },
  ".jhipster/FieldTestServiceImplEntity.json": {
    "stateCleared": "modified",
  },
  ".jhipster/JpaFilteringOtherSide.json": {
    "stateCleared": "modified",
  },
  ".jhipster/JpaFilteringRelationship.json": {
    "stateCleared": "modified",
  },
  ".jhipster/Label.json": {
    "stateCleared": "modified",
  },
  ".jhipster/MapsIdChildEntityWithDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/MapsIdChildEntityWithoutDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/MapsIdGrandchildEntityWithDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/MapsIdGrandchildEntityWithoutDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/MapsIdParentEntityWithDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/MapsIdParentEntityWithoutDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/MapsIdUserProfileWithDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/Operation.json": {
    "stateCleared": "modified",
  },
  ".jhipster/Place.json": {
    "stateCleared": "modified",
  },
  ".jhipster/SuperMegaLargeTestEntity.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestCustomTableName.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestEntity.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestInfiniteScroll.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestManyRelPaginDTO.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestManyToMany.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestManyToOne.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestMapstruct.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestOneToOne.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestPagination.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestServiceClass.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestServiceImpl.json": {
    "stateCleared": "modified",
  },
  ".jhipster/TestTwoRelationshipsSameEntity.json": {
    "stateCleared": "modified",
  },
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
}
`;

exports[`generator - generate-sample with ng-default-additional (star jdl-entity) should match matrix value 1`] = `
{
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "01-non-updatable-entity.jdl": {
    "stateCleared": "modified",
  },
  "02-authority-relationship.jdl": {
    "stateCleared": "modified",
  },
  "custom-domain.jdl": {
    "stateCleared": "modified",
  },
  "custom-id.jdl": {
    "stateCleared": "modified",
  },
  "cyclic.jdl": {
    "stateCleared": "modified",
  },
  "entities.jdl": {
    "stateCleared": "modified",
  },
}
`;

exports[`generator - generate-sample with ng-webflux-psql-additional (specific jdl-samples) should match matrix value 1`] = `
{
  "custom-domain.jdl": {
    "stateCleared": "modified",
  },
  "webflux-psql.jdl": {
    "stateCleared": "modified",
  },
}
`;

exports[`generator - generate-sample with vue-default-additional (specific jdl-entity) should match matrix value 1`] = `
{
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "custom-id.jdl": {
    "stateCleared": "modified",
  },
}
`;
